Fix docker env and improve scripts

patcon opened the pull request:
Resolves https://github.com/GCTC-NTGC/gc-digital-talent/issues/1782

This PR:
- fixes the php-curl missing from docker that was preventing `composer install`, 05594cc53cd40fe98add35f50369123ceee012b7
- adds error-handling to scripts, raising errors, pipefails, and undeclared vars 27cdc7f078b25e6e6ee6ef9e2dd0e610ba18d758
- makes some fixes to scripts that were detected 
  - makes scripts more flexible (run in docker or host machine) 373eeef79bc272784c313b4427d15befd524ac6b
  - cross-platform random string generation, now osx+linux (prev way was also generating pipefail) 2922bc5830b51d456b3e749175efc0df0556f03a 2bfe9ae83b611f9d64a7f44821b199140cb856f6

patcon commented:
k builds fine! Ready for review! cc @yonikid15 @petertgiles

petertgiles commented:
I love where this is going.  Getting the php-curl in is important and the shell options to be able to notice the errors are so important.  I like the lib subfolder for scripts too.  Maybe the update*.sh scripts should get moved under there, too?

I understand what you're doing with parent_path and I love it because that's a big pain point for me too.  I feel like you're 95% of the way there.  Is there a way to remove the hardcoded paths to `/var/www/html/`?  As long as those are there you can't really run the scripts on your host machine unless you've manually symlinked that path first, right?  When I was fiddling with this I was working on an environment variable for something like WORKSPACE_ROOT but I couldn't figure out where to set it.  When running in docker that's easy enough but how do you set it reliably on your host machine?  I bet you've already got a great idea on a different way to handle this!

Cool update to the random appkey generator.  If it's not too much to explain, could you help me understand the issue and the change?  I don't have access to a mac to try it out but it does seem to still work on linux.

patcon commented:
> If it's not too much to explain, could you help me understand the issue and the change?

Sure! I added `set -o pipefail` so that error codes in the middle of piped commands still surface. But due to `cat /dev/urandom` being an infinite stream that gets cut off, it will always raise a pipefail. The command I added is another approach to generate randomness that doesn't cut off a pipe like that ([ref](https://unix.stackexchange.com/questions/182382/generating-a-random-password-why-isnt-this-portable)). (Alternatively, we could keep what we had, but `set +o pipefail` right before, and turn it back on after.)

There are a few ways to generate randomness, but I opted for `dd` because it's available in docker, ubuntu, and osx (whereas `hexdump` is missing from docker and some others aren't portable to osx).

---

And happy to make the other changes later today :)

patcon commented on maintenance/scripts/update_env_appkey.sh:
```diff
@@ -1,3 +1,8 @@
+parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+source ${parent_path}/lib/common.sh
+
+# Random string generation that works on OSX and Linux. (Plus not pipefails.)
+appkey=`dd if=/dev/urandom bs=48 count=1 2>/dev/null | base64 | tr -dc '[a-zA-Z0-9]' | fold -w32 -- | head -n 1`
```
```suggestion
# Random string generation that works with builtin utils on OSX and Linux. (Plus not pipefails.)
# See: https://unix.stackexchange.com/a/182484
appkey=`dd if=/dev/urandom bs=48 count=1 2>/dev/null | base64 | tr -dc '[a-zA-Z0-9]' | fold -w32 -- | head -n 1`
```

patcon commented:
> Maybe the update*.sh scripts should get moved under there, too?

Ah good call. So those aren't intended to be run by us, just other scripts?

patcon commented:
Is it possible for me to make those "/var/www/html" changes in a small PR after merging this? I think it's a great idea, and should be simple, but will prob conflict with stuff in #1798. And I'm hoping to make it as easy as possible to merge that, since that PR gets complicated if its not merged soon

petertgiles reviewed (approved):
This is great stuff, Pat, thanks.
